src/context.rs

```diff
@@ -431,7 +431,16 @@ fn get_config_path_os(env: &Env) -> Option<OsString> {
     if let Some(config_path) = env.get_env_os("STARSHIP_CONFIG") {
         return Some(config_path);
     }
-    Some(home_dir(env)?.join(".config").join("starship.toml").into())
+
+    let config_dir = home_dir(env)?.join(".config");
+    let config_file_path = config_dir.join("starship.toml");
+    let folder_based_config_file_path = config_dir.join("starship/config.toml");
```

Comment:
For this, I would prefer following the XDG-spec (and things like `$XDG_CONFIG_HOME`), or use the platform-native directory via `dirs`. Please consult the discussions in #896.

Reply:
I've updated to follow the XDG-spec. It should respect the `$XDG_CONFIG_HOME` and fallback to `$HOME/.config` if the env is not set. 
I'm not sure if it could break current user config in case they have `$XDG_CONFIG_HOME` in different place but `starship.toml` still in `~/.config/`.

Reply:
I think only `~/.config/starship/config.toml` should use `$XDG_CONFIG_HOME` and `~/.config/starship.toml` should only use `$HOME` to keep compatibility.